lib/katello/tasks/repository.rake

```diff
@@ -108,7 +108,7 @@ namespace :katello do
       return false unless backend_service&.repository_reference&.repository_href
       backend_service.api.repositories_api.read(backend_service.repository_reference.repository_href)
     else
-      Katello.pulp_server.extensions.repository.retrieve(repo.pulp_id)
```

Comment:
If I'm remembering right, this line would trigger a 404 error and get caught below in `rescue StandardError`.  The false there would get replaced by the true on line 113 as the return object in the odd case that Pulp 3 isn't supported.

I'd replace the entire method with something like:

```ruby
def repo_exists?(repo)
  backend_service = repo.backend_service(SmartProxy.pulp_primary!)
  return false unless backend_service&.repository_reference&.repository_href
  backend_service.api.repositories_api.read(backend_service.repository_reference.repository_href)
  true
rescue StandardError => e
  return false if e.code == 404
end
```